operator/validator/event_handler.go

```diff
@@ -141,30 +142,25 @@ func (c *controller) handleValidatorAddedEvent(
 	logger *zap.Logger,
 	event abiparser.ValidatorAddedEvent,
 	ongoingSync bool,
-) ([]zap.Field, error) {
-	_, found, err := c.eventHandler.GetEventData(event.TxHash)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get event data")
+) (logFields []zap.Field, err error) {
+	var valid bool
+	defer func() {
+		err = c.handleValidatorAddedEventDefer(valid, err, event)
```

Comment:
I rather the code be here than jumping to a different func. I know it looks a bit less clean but its easier to follow.